iothub/service/src/Exceptions/IotHubServiceException.cs

```diff
@@ -104,6 +104,14 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
             info.AddValue(TrackingIdValueSerializationStoreName, TrackingId);
         }
 
+        /// <inheritdoc/>
+        public override string ToString()
+        {
+            return $"Message: {Message}\n" +
```

Comment:
Suggested format would be:

$"{Message}\nError code={ErrorCode}, ...\n{StackTrace}"